tidy3d/components/geometry.py

```diff
@@ -2069,67 +2115,66 @@ def from_gds(  # pylint:disable=too-many-arguments, too-many-locals
         polys_union = functools.reduce(lambda poly1, poly2: poly1.union(poly2), polygons)
 
         if isinstance(polys_union, Polygon):
-            all_vertices = [cls.strip_coords(polys_union)[0]]
+            all_vertices = [PolySlab.strip_coords(polys_union)[0]]
         elif isinstance(polys_union, MultiPolygon):
-            all_vertices = [cls.strip_coords(polygon)[0] for polygon in polys_union.geoms]
-
-        return [
-            cls(
-                vertices=verts,
-                axis=axis,
-                slab_bounds=slab_bounds,
-                dilation=dilation,
-                sidewall_angle=sidewall_angle,
-                reference_plane=reference_plane,
-                **kwargs,
-            )
-            for verts in all_vertices
-        ]
+            all_vertices = [PolySlab.strip_coords(polygon)[0] for polygon in polys_union.geoms]
+        return all_vertices
 
     @cached_property
-    def base_polygon(self) -> Vertices:
-        """The polygon at the base after potential dilation operation.
-        The vertices will always be transformed to be "proper".
+    def reference_polygon(self) -> np.ndarray:
+        """The polygon at the reference plane.
 
         Returns
         -------
         ArrayLike[float, float]
-            The vertices of the polygon at the base.
+            The vertices of the polygon at the reference plane.
         """
-
-        # offset value to obtain polygon in the base
-        offset_base = self.dilation + self.offset_distance_to_base(
-            self.reference_plane, self.length_axis, self._tanq
-        )
-        return self._shift_vertices(self._proper_vertices(self.vertices), offset_base)[0]
+        vertices = self._proper_vertices(self.vertices)
+        if isclose(self.dilation, 0):
+            return vertices
+        return self._shift_vertices(vertices, self.dilation)[0]
```

Comment:
This dilation/erosion might itself result in self-intersecting polygons. Should we do some clean-up to remove those? Here the problem is easier than with the extrusion, since it doesn't have to be compared to any other slabs. @weiliangjin2021 I believe you mentioned that shapely had already a function that could be used for this, right?

Reply:
I was thinking about doing this as well, but then quickly realized that if we allow self-intersecting polygon at the reference plane and perform clean-up, there are situations where the healed polygon can split and have islands/holes. So right now, we simply error in this case: https://github.com/flexcompute/tidy3d/blob/94b801b6c4f71c6c73a52c3622cecb90e8c07707/tidy3d/components/geometry.py#L1854-L1855

Another possibility is to error only when the healed polygon splits or generates holes/islands, but allow the change of vertices number. What do you think?

Reply:
I think its usually best to make it easier for the user, as long as we don't overstep on what their intentions might be. In this case, we should gracefully handle both cases, but I can see that the current API might make it difficult to return multiple islands. Maybe continue to error in this case, but handle the easier one, where we just clean-up the polygon and remain with a single proper polygon?

Reply:
Shall we error or just warning for polygon splitting/islands? I'm thinking about: 1) error for polygon splitting/islands; 2) warning for the change of the number of vertices.

Reply:
This part has been added. Quite significant change in the two validators.

Reply:
Actually, what I have now doesn't really address your comments, since I still offset the vertices first and then heal. I cannot think of any self-intersecting polygon that can really pass the validator here. Maybe let's plan on the next release.